forms_pro/api/export.py

```
@@ -0,0 +1,104 @@
+        # CSV reads `frappe.response["doctype"]` (via `as_csv`); Excel reads
+        # `frappe.response["filename"]` (via `provide_binary_file`). Set both.
+        safe_title = secure_filename(form.title or "") or form_id
+        timestamp = frappe.utils.now_datetime().strftime("%Y-%m-%d_%H%M%S")
+        base_name = f"Submissions_{safe_title}_{timestamp}"
+        frappe.response["doctype"] = base_name
+        if file_type == "Excel":
+            frappe.response["filename"] = f"{base_name}.xlsx"
+    finally:
+        try:
+            frappe.set_user(saved_user)
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)